client/dashboard/app/queries.ts

```diff
@@ -240,3 +244,41 @@ export function siteDefensiveModeMutation( siteSlug: string ) {
 		},
 	};
 }
+
+export function siteHasPurchasesCancelableQuery( siteId: string, userId: number ) {
+	return {
+		queryKey: [ 'site', siteId, 'purchases-cancelable' ],
+		queryFn: () => {
+			return fetchPurchases( siteId );
+		},
+		select: ( purchases: Purchase[] ) => {
+			const cancelables = purchases
+				.filter( ( purchase ) => {
+					// Exclude inactive purchases and legacy premium theme purchases.
+					if ( ! purchase.active || purchase.product_slug === 'premium_theme' ) {
+						return false;
+					}
+
+					return purchase.is_cancelable;
+				} )
+				.filter( ( purchase ) => Number( purchase.user_id ) === userId );
```

Comment:
Would it be better to make it optional so we can tell whether there are active purchases on the site or for the user?

Reply:
I think we can update it when we need to implement the use case 🙂